src/main/java/com/google/devtools/build/lib/runtime/UiEventHandler.java

```diff
@@ -207,7 +198,7 @@ public UiEventHandler(
               Math.round(options.showProgressRateLimit * 1000),
               NO_CURSES_MINIMAL_PROGRESS_RATE_LIMIT);
     }
-    this.minimalUpdateInterval = Math.max(this.minimalDelayMillis, MAXIMAL_UPDATE_DELAY_MILLIS);
+    this.minimalUpdateInterval = Math.max(this.minimalDelayMillis, MINIMAL_UPDATE_INTERVAL_MILLIS);
```

Comment:
I believe that could be risky. [`doRefresh`](https://github.com/bazelbuild/bazel/blob/862965a209195a6d8bbefed17e8a1d283753e82c/src/main/java/com/google/devtools/build/lib/runtime/UiEventHandler.java#L846)  respects `minimalDelayMillis`, which can be very small (if `--show_progress_rate_limit` is set to a very low value because the user wants to see many events). If we'd use `minimalUpdateInterval` instead, we could get at most 1 update every 0.2s (the value of `MINIMAL_UPDATE_INTERVAL_MILLIS`, so users could get fewer updates than expected.

We also can't trivially remove `MINIMAL_UPDATE_INTERVAL_MILLIS`, because then the [update thread](https://github.com/bazelbuild/bazel/blob/862965a209195a6d8bbefed17e8a1d283753e82c/src/main/java/com/google/devtools/build/lib/runtime/UiEventHandler.java#L924) would continuously awaken.

But I agree it's annoying to have two very similar instance variables. Maybe a solution could be to put a lower minimum (for example 0.01s)? Or would that still make the update thread too heavy? Alternatively, maybe it makes sense to just rename `minimalDelayMillis` to `progressRateLimit` to highlight the link with the `--show_progress_rate_limit` flag?

Reply:
Good point, I'd rather not change any other behavior at this stage.

I agree with your latter suggestion to change it to something like `progressRateLimitMillis`.